synapse/python_dependencies.py

```diff
@@ -107,7 +107,7 @@
     "url_preview": ["lxml>=3.5.0"],
     "sentry": ["sentry-sdk>=0.7.2"],
     "opentracing": ["jaeger-client>=4.0.0", "opentracing>=2.2.0"],
-    "jwt": ["pyjwt>=1.6.4"],
+    "jwt": ["pyjwt>=2.1.0"],
```

Comment:
I asked the packages whether this change would be OK. Can you remind me why we're bumping this?

Reply:
Looks like supported versions of Debian and Fedora are on 1.7.1 still. Would it be possible to use that?

Reply:
`PyJWKClient` was added in v2.0.0 and caching of signing keys is supported since v2.1.0: https://github.com/jpadilla/pyjwt/blob/master/CHANGELOG.rst#v210

Reply:
I'm not really sure what to do about this, although I do wonder if we should be using `PyJWKClient` since it bypasses the reactor / current methods for communicating with external resources.